### ISSUE TITLE
Pytest always outputs traceback for XFAIL [NOTRUN] test

### ISSUE THREAD

**Issue by Alex-Bogdanov** [pytest-dev/pytest-html#118]
Hi!
Is there anybody who can elaborate what is going wrong with _pytest-html_, which generates results for **XFailed** tests on _'setup'_ without running the actual test? It is always printing this traceback in details:
`self = <CallInfo when='setup' exception: [NOTRUN] Failed due to changes in functionality>, func = <function call_runtest_hook.<locals>.<lambda> at 0x7fc6c50d9f28>, when = 'setup'

    def __init__(self, func, when):
        #: context of invocation: one of "setup", "call",
        #: "teardown", "memocollect"
        self.when = when
        self.start = time()
        try:
>           self.result = func()

.venv/lib/python3.5/site-packages/_pytest/runner.py:163: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.venv/lib/python3.5/site-packages/_pytest/runner.py:151: in <lambda>
    return CallInfo(lambda: ihook(item=item, **kwds), when=when)
.venv/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py:745: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
.venv/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py:339: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
.venv/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py:334: in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
.venv/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py:613: in execute
    return _wrapped_call(hook_impl.function(*args), self.execute)
.venv/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py:254: in _wrapped_call
    return call_outcome.get_result()
.venv/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py:279: in get_result
    raise ex[1].with_traceback(ex[2])
.venv/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py:265: in __init__
    self.result = func()
.venv/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py:614: in execute
    res = hook_impl.function(*args)
.venv/lib/python3.5/site-packages/_pytest/skipping.py:190: in pytest_runtest_setup
    check_xfail_no_run(item)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

item = <Function 'test_send_sync_with_distinct_sessions'>

    def check_xfail_no_run(item):
        """check xfail(run=False)"""
        if not item.config.option.runxfail:
            evalxfail = item._evalxfail
            if evalxfail.istrue():
                if not evalxfail.get('run', True):
>                   pytest.xfail("[NOTRUN] " + evalxfail.getexplanation())
E                   _pytest.skipping.XFailed: [NOTRUN] Failed due to changes in functionality

.venv/lib/python3.5/site-packages/_pytest/skipping.py:208: XFailed`

Mainly, I just wanna to see only the reason, which is `XFailed: [NOTRUN] Failed due to changes in functionality`, same as it works for **SKIPPED** tests

Thanks!

**Comment by davehunt** [comment]
Could you provide more context, or a minimised test case that reproduces this issue? It would also be useful to know what versions of pytest and the plugins you're using. A full console log would include this.

**Comment by davehunt** [comment]
Wait, are you saying that the traceback is appearing in the HTML report for an xfailed test that is marked to not run? If so, I agree that it makes sense to not output this in the report. I'll label this as an enhancement request, and maybe someone will be able to work on it soon.

**Comment by Alex-Bogdanov** [comment]
Yep! Exactly. So, since I've posted it, maybe I can rewrite this code, generating html output for xfailed tests marked not to run?=)

**Comment by Alex-Bogdanov** [comment]
Looks like it is pytest code base issue, not plugin

**Comment by davehunt** [comment]
> Looks like it is pytest code base issue, not plugin

Have you raised an issue against the pytest project? Could we workaround the issue in the plugin?

**Comment by nicoddemus** [comment]
@Alex-Bogdanov if you could reproduce this problem in a simple case and post to https://github.com/pytest-dev/pytest/issues it would be great!

**Comment by Alex-Bogdanov** [comment]
@nicoddemus I will try to reproduce without html plugin, it's god damn tricky
